### PR TITLE
chore(imprvements): code changes for fileUploader

### PR DESCRIPTION
**Pull request by saadjutt01**
## Issue

No issue to link

## Intent

Improve code and stick to standards followed in other sections of codebase. 

## Implementation

What code changes have been made to achieve the intent.

## Checks

No PR (that involves a non-trivial code change) should be merged, unless all items below are confirmed!  If an urgent fix is needed - use a tar file.

- [x] All `sasjs-cli` unit tests are passing (`npm test`).
Mocked:
image
Server:
image
- [x] All `sasjs-tests` are passing (instructions available [here](https://github.com/sasjs/adapter/blob/master/sasjs-tests/README.md)).
VIYA:
image
SAS9:
- [x] [Data Controller](https://datacontroller.io) builds and is functional on both SAS 9 and Viya
VIYA:
image
SAS9:
image

### CODE REVIEW

File: src/FileUploader.ts [hunk 1]
```diff
@@ -3,21 +3,17 @@ import { UploadFile } from './types/UploadFile'
...
 
 export class FileUploader {
-  constructor(
-    private sasjsConfig: SASjsConfig,
-    private jobsPath: string,
-    private requestClient: RequestClient
-  ) {
-    if (this.sasjsConfig.serverUrl) isUrl(this.sasjsConfig.serverUrl)
-  }
+  constructor(private jobsPath: string, private requestClient: RequestClient) {}
```
Comment (saadjutt01) [inline src/FileUploader.ts:9]: Removed `sasjsConfig`, as private member of `FileUploader`. We already have sasjsConfig in `SASjs.ts`

Looking at the usage of file uploading in fileUploader repo, calling it through `SASjs` instance
- [usage 1](https://github.com/sasjs/fileuploader/blob/f7959bef0f192fd05b6ba2074dbea8758b0f6156/src/scripts.js#L64)
- [usage 2](https://github.com/sasjs/fileuploader/blob/f7959bef0f192fd05b6ba2074dbea8758b0f6156/src/scripts.js#L91)

File: src/FileUploader.ts [hunk 2]
```diff
@@ -3,21 +3,17 @@ import { UploadFile } from './types/UploadFile'
...
 
-  public uploadFile(sasJob: string, files: UploadFile[], params: any) {
+  public async uploadFile(
+    sasJob: string,
+    files: UploadFile[],
+    params: any,
+    config: SASjsConfig
```
Comment (saadjutt01) [inline src/FileUploader.ts:15]: Passing in `SASjsConfig`, would keep it same as other `JobExecutor` code files.

File: src/SASjs.ts [hunk 3]
```diff
@@ -573,22 +573,16 @@ export default class SASjs {
...
     overrideSasjsConfig?: any
   ) {
-    const fileUploader = overrideSasjsConfig
-      ? new FileUploader(
-          { ...this.sasjsConfig, ...overrideSasjsConfig },
-          this.jobsPath,
-          this.requestClient!
-        )
-      : this.fileUploader ||
-        new FileUploader(this.sasjsConfig, this.jobsPath, this.requestClient!)
```
Comment (saadjutt01) [inline src/SASjs.ts:589]: It's not helpful to create new `FileUploader` instances. We already have one ( instantiated through `SASjs` constructor) and will always be present.

File: src/SASjs.ts [hunk 4]
```diff
@@ -990,11 +984,7 @@ export default class SASjs {
         )
     }
 
-    this.fileUploader = new FileUploader(
-      this.sasjsConfig,
-      this.jobsPath,
-      this.requestClient
-    )
+    this.fileUploader = new FileUploader(this.jobsPath, this.requestClient)
```
Comment (saadjutt01) [inline src/SASjs.ts:987]: `FileUploader` doesn't have to maintain the state of `SASjsConfig`

File: src/job-execution/JobExecutor.ts [hunk 5]
```diff
@@ -21,7 +20,6 @@ export abstract class BaseJobExecutor implements JobExecutor {
   constructor(protected serverUrl: string, protected serverType: ServerType) {}
 
   private waitingRequests: ExecuteFunction[] = []
-  private requests: SASjsRequest[] = []
```
Comment (saadjutt01) [inline src/job-execution/JobExecutor.ts:24]: residue from recent code changes.

**Review by krishna-acondy — APPROVED** [review]

**Review by sabhas — APPROVED** [review]

**Review by VladislavParhomchik — APPROVED** [review]

**Pull request merged**